### DISCUSSION TITLE
Https connection?

### DISCUSSION THREAD

**Discussion by D3JA5U** [LivingWithHippos/unchained-android#225] [General]
Hi, I only recently discovered real-debrid services and your app. I tried unchained on tablet and phone - it is working really well. 

I am wondering about two things:
- are all established app connections (search sites, real-debrid) via https?
- real-debrid has a warning to be careful where to use the API token. Is login info/API stored/transmitted in a secure way in your app?

Your readme does not have any info on this. Might be helpful to add something.

Again, I am using your app (with VPN) and I like it a lot. 
Hope your user base will grow and keeps you going a long way with this.
Thank you!

**Comment by LivingWithHippos** [comment]
> are all established app connections (search sites, real-debrid) via https?

- All the connections with real debrid are with https.
- All the plugins except max-rls and scene-rls are https only. They didn't have https support at the time but it seems they added it now so I'll update them soon if they still work, because they are adding cloudflare which stops the necessary scraping. 
- Https is not magic anyway, people can't intercept the call content but stuff like the address requested should still be visible (if I remember correctly)

> real-debrid has a warning to be careful where to use the API token. Is login info/API stored/transmitted in a secure way in your app?

The notice is 
> Warning: Never ever use your private API token for public applications, it is insecure and gives access to all methods.

With public applications they mean the ones shared with other people, not like Unchained which is supposed to be used by yourself only. You can't "expire" the private token to remove its access, while with the open one you can, see the device page https://real-debrid.com/devices

Anyway you can just use the open token, the only feature missing will be /transcoding. I can't find documentation on this anymore but with the open token it returns  4 | Method not allowed

> Is login info/API stored/transmitted in a secure way in your app?

Transmitted yes. Stored more or less, it's in the app internal memory which is not accessible from other apps. An attacker would need to either root and hack your phone or steal it so I wasn't particularly worried about that since you probably would have more worries than real debrid then. I suppose I could add encryption to it sooner or later but I'm not in a hurry.
